src/App.js

```diff
@@ -18,27 +24,78 @@ const theme = createMuiTheme({
   }
 });
 
-export default function App() {
+const AuthRoute = props => {
+  if (!props.user.roles.includes('UNAUTHORIZED')) return <Redirect to="/" />;
+  return <Route {...props} />;
+};
```

Comment:
I think we could have AuthUserRoute and AuthManagerRoute and that would be sufficient for our needs. So just extending this a bit would suit all our needs, I think?

Reply:
And I think that the default Unauthorized redirect should be to login page.

Reply:
I think we can do just one 'PrivateRoute' or 'AuthRoute' I don't know how to call it, to which you would send the roles, which can go to that page. For example to Main page we can send roles - ['DEVELOPER', 'LEADER'] as a props, and then in this 'AuthRoute' we can check if user have that role, if yes, then redirect to Main page, if no to login, or something like this if you get what I mean. I don't think that we need separate Route components for all the roles.

Reply:
And in this case I check if user don't have role unauthorized, I redirect him to MainPage, because logged in user should not have possibility to visit login page.

Reply:
Yes, that makes more sense. And then the MainPage should take care of routing to the login page If needed.